src/Nethermind/Nethermind.Synchronization/StateSync/StateSyncDispatcher.cs

```diff
@@ -49,7 +86,82 @@ protected override async Task Dispatch(PeerInfo peerInfo, StateSyncBatch request
                     {
                         batchLocal.Responses = t.Result;
                     }
-                }, request);
+                }, batch);
+        }
+
+        /// <summary>
+        /// SNAP protocol allows grouping of storage requests by account path.
+        /// The grouping decrease requests size.
+        /// </summary>
+        private GetTrieNodesRequest GetGroupedRequest(StateSyncBatch batch)
```

Comment:
Some questions:
1. This looks like a lot of allocations, can we reduce them?
2. Shouldn't we group this not here at the level of sending the request, but at higher level of traversing the tree and keeping StateSyncItems? There this grouping should come up naturally.